lib/postprocessing/image.js

```diff
@@ -142,9 +191,45 @@ function hasJpegQuality(metadata, instructions) {
     return true;
 }
 
+/**
+ * Returns updated rendition instructions for the worker callback processing if needed
+ * or undefined if no changes are required.
+ */
+async function prepareImagePostProcess(rendition, options) {
+    if (rendition.instructions.fmt && isSupportedOutputFormat(rendition.instructions)) {
+
+        const supportedFormats =
+            options.supportedRenditionFormats
+            || WORKER_SUPPORTED_FMT[new OpenwhiskActionName().name]
+            || DEFAULT_WORKER_SUPPORTED_FMT;
+
+        // in case the worker cannot generate the final rendition format itself, we have to do it via post processing
+        if (!supportedFormats.includes(rendition.instructions.fmt)) {
+            // ...and tell the worker to generate a different intermediate format it can do
+            // find first format that is supported (worker defines priority order)
+            const newFormat = supportedFormats.find(fmt => SUPPORTED_INPUT_FMT.includes(fmt));
+            if (!newFormat) {
+                throw new GenericError(`Worker does not support generating a compatible image format. Requested: ${SUPPORTED_INPUT_FMT} Supported by worker: ${supportedFormats}`);
+            }
+
+            console.log(`post-processing: adjusted fmt for worker callback: '${rendition.instructions.fmt}' -> '${newFormat}'`);
+
+            rendition._forceImagePostProcess = true;
+
+            return Object.assign({}, rendition.instructions, { fmt: newFormat});
+        }
+    }
+
+    // return undefined if no instruction change needed
+}
+
+/**
+ * Returns true if the (intermediate) rendition needs image post processing.
+ */
 async function needsImagePostProcess(rendition) {
     const instructions = rendition.instructions;
 
+    // NOTE: already checked in prepareImagePostProcess() above, no need to check again?
```

Comment:
+1 this check seems repetitive now (honestly more confusing with it still here)

Reply:
I think we still need it, since `prepareImagePostProcess()` never _denies_ post processing. It just makes adjustments to the instructions. This is actually the common case for all non-image renditions.

For example, let's say we have `fmt=xmp`. `prepareImagePostProcess()` will do nothing, and then when we get to `needsImagePostProcess()` here this is the place where we would decide to do nothing, since xmp is not something image post processing would work on.

Will remove the comment, which is indeed wrong.